docs/source/python/orc.rst

```diff
@@ -0,0 +1,210 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _orc:
+
+Reading and Writing the Apache ORC Format
+=============================================
+
+The `Apache ORC <http://orc.apache.org/>`_ project provides a
+standardized open-source columnar storage format for use in data analysis
+systems. It was created originally for use in `Apache Hadoop
+<http://hadoop.apache.org/>`_ with systems like `Apache Drill
+<http://drill.apache.org>`_, `Apache Hive <http://hive.apache.org>`_, `Apache
+Impala (incubating) <http://impala.apache.org>`_, and `Apache Spark
+<http://spark.apache.org>`_ adopting it as a shared standard for high
+performance data IO.
+
+Apache Arrow is an ideal in-memory representation layer for data that is being read
+or written with ORC files.
+
+Obtaining pyarrow with ORC Support
+--------------------------------------
+
+If you installed ``pyarrow`` with pip or conda, it should be built with ORC
+support bundled:
+
+.. ipython:: python
+
+   import pyarrow.orc as po
+
+If you are building ``pyarrow`` from source, you must use
+``-DARROW_ORC=ON`` when compiling the C++ libraries and enable the ORC
+extensions when building ``pyarrow``. See the :ref:`Python Development
+<python-development>` page for more details.
+
+Reading and Writing Single Files
+--------------------------------
+
+The functions :func:`~.orc.read_table` and :func:`~.orc.write_table`
+read and write the :ref:`pyarrow.Table <data.table>` object, respectively.
+
+Let's look at a simple table:
+
+.. ipython:: python
+
+   import numpy as np
+   import pandas as pd
+   import pyarrow as pa
+
+   df = pd.DataFrame({'one': [-1, np.nan, 2.5],
+                      'two': ['foo', 'bar', 'baz'],
+                      'three': [True, False, True]},
+                      index=list('abc'))
```

Comment:
I would leave out the custom index here, to have the example focus on reading and writing ORC, and not on the pandas<->arrow conversion details. 
Maybe we could even start with creating directly a pyarrow.Table from this dict (`pa.table({..})` instead of `pd.DataFrame({{..})`)?

Reply:
Yup.